### PR TITLE
scan: search bar on validator home page

### LINKED ISSUE #1896: Search bar on Validator Home Page
[Ref](https://www.mintscan.io/validators)

Tasks
- Input UI
- Search functionality

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1896

### CODE REVIEW

**Comment by evilpeach** [comment]
@tansawit  Please fix the conflict.

**Review by evilpeach — CHANGES_REQUESTED** [review]
Please the conflict.

File: scan/src/pages/ValidatorHomePage.re [hunk 1]
```diff
@@ -430,12 +440,17 @@ module ValidatorList = {
...
+               });
          <>
-           {validators->Belt_Array.size > 0
-              ? validators
-                ->sorting(sortedBy)
-                ->Belt_Array.map(e =>
-                    renderBody(e.rank, Sub.resolve(e), bondedTokenCount.amount)
+           {filteredValidator->Belt_Array.size > 0
+              ? filteredValidator
+                ->Belt_Array.mapWithIndex((i, e) =>
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: sorting ??

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
IMO, we should disable the toggle button by using opacity.

File: scan/src/pages/ValidatorHomePage.re [hunk 2]
```diff
@@ -596,7 +612,18 @@ let make = () => {
         </Col>
       </Row>
     </div>
-    <ValidatorList allSub />
+    <div className=Styles.searchContainer>
+      <input
+        type_="text"
+        className=Styles.searchBar
+        placeholder="Search Validator"
+        onChange={event => setSearchTerm(ReactEvent.Form.target(event)##value)}
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: setState function expects the function that returns a new state.
So, It should be like this.
```
event => {
      let newVal = ReactEvent.Form.target(event)##value;
      setSearchTerm(_ => newVal);
}
```

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**